webapp/channels/src/components/cloud_preview_modal/cloud_preview_modal.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {getCloudSubscription} from 'mattermost-redux/selectors/entities/cloud';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+
+import {modalContent} from './preview_modal_content_data';
+import PreviewModalController from './preview_modal_controller';
+
+const CLOUD_PREVIEW_MODAL_SHOWN_PREF = 'cloud_preview_modal_shown';
+
+const CloudPreviewModal: React.FC = () => {
```

Comment:
Nit: I'm realizing that the names for this and the controller are sort of inverted because this outer component controls whether or not the modal is shown

Reply:
I see what you're saying, but in this context by `controller` I meant it controls the functionality of the modal itself (ie, the slide show, the buttons within it etc). I'm open to other names but this its probably not a big deal to leave it?

Reply:
Typically, the various `WhateverController` components in the web app are either high level components in charge of loading and routing through parts of the app (`TeamController`, `ChannelController`, `LoginController`) or for determining when other elements should be mounted (`ModalController`, `AnnouncementBarController`). With the current naming, when I see that the `LoggedIn` component is rendering a `WhateverModal`, the first thing I think is "Why are we mounting a modal there? It shouldn't be mounted all the time."

Could you change those names around?